File: java/linkedlist/Stack.java
public class Stack {
    static class Node{
        int data;
        Node next;
        public Node(int data){
            this.data = data;
        }
    }
    public Node top;
    public void push(int data){
        Node newNode = new Node(data);
        if(top==null){
            top = newNode;
        }else{
            newNode.next = top;
            top = newNode;
        }
    }
    public void pop(){
        if(top==null){
            System.out.println("stack underflow");
            return ;
        }
        top = top.next;
    }
    public void display(){
        Node current = top;
        while(current!=null){
            System.out.println(current.data);
            current = current.next;
        }
    }
    public static void main(String[] args) {
        Stack s = new Stack();
        s.push(20);
        s.push(34);
        s.push(66);
        s.display();
        System.out.println("\n");
        s.pop();
        s.display();
    }
}

File: c++/linkedlist/stack.cpp
#include <bits/stdc++.h>
using namespace std;
struct Node{
	int data;
	Node* link;
};
Node* top;
void push(int data){
	Node* temp = new Node();
	if(!temp){
		cout << "\nStack Overflow";
		exit(1);
	}
	temp->data = data;
	temp->link = top;
	top = temp;
}
int isEmpty(){
	return top == NULL;
}
int peek(){
	if (!isEmpty())
		return top->data;
	else
		exit(1);
}
void pop(){
	Node* temp;
	if (top == NULL){
		cout << "\nStack Underflow" << endl;
		exit(1);
	}
	else{
		temp = top;
		top = top->link;
		free(temp);
	}
}
void display(){
	Node* temp;
	if (top == NULL){
		cout << "\nStack Underflow";
		exit(1);
	}
	else{
		temp = top;
		while (temp != NULL){
			cout << temp->data << " -> ";
			temp = temp->link;
		}
	}
}
int main(){
	push(11);
	push(22);
	push(33);
	push(44);
	display();
	cout << "\nTop element is " << peek() << endl;
	pop();
	pop();
	display();
	cout << "\nTop element is " << peek() << endl;
	return 0;
}
